_playwright-tests/UI/CustomRepoCRUD.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+
+export const repoNamePrefix = 'Repo-CRUD';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+test.describe('Custom Repositories CRUD', () => {
+  test('Add, Read, update, delete a repo', async ({ page }) => {
+    await test.step('Delete any CRUD test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      const repositoryModal = page.locator('div[id^="pf-modal-part"]').first();
+      await expect(repositoryModal).toBeVisible();
+      await expect(page.locator('text="Add custom repositories"')).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save' }).first().click();
```

Comment:
Are there multiple save buttons? :thinking:

Reply:
No, I'll remove all instances of "first" and test again.

Reply:
Actually there are two. Best way would be to use the test ID `modal_save`:
`      await page.getByTestId('modal_save').click();
`